### ISSUE TITLE
LoginAsync triggering exception while debugging VS2017

### ISSUE THREAD

**Issue by pdefreitas** [Microsoft/winsdkfb#237]
Debug mode: Managed + Native (mixed).

Right after a await LoginAsync an exception is thrown:
Exception thrown at 0x750DB782 in Demo.exe: Microsoft C++ exception: Platform::COMException ^ at memory location 0x067FF2B0. HRESULT:0x80070002
excptptr.cpp line 543.

ThisException	{ExceptionCode=3765269347 ExceptionFlags=1 ExceptionRecord=0x00000000 <NULL> ...}	_EXCEPTION_RECORD
+		ExceptionInformation	0x067ff2ec {429065504, 109048496, 192996064, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0}	unsigned long[15]
ThisException.ExceptionFlags	1	unsigned long
ThisException.ExceptionInformation	0x067ff2ec {429065504, 109048496, 192996064, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0}	unsigned long[15]
ThisException.NumberParameters	3	unsigned long

Ps: I'm not doing anything fancier, just following up the documentation doing an async button click event for login. Without native debugging no exception is thrown.

**Comment by adiviness** [comment]
Does this exception prevent the app from logging in? Visual studio is really eager with telling you about exceptions by default, it will warn you even if the exception will be handled by a try/catch block. We do expect exceptions to be created at certain points in the login process (and in other scenarios) but they should all be handled correctly by the code.

**Comment by pdefreitas** [comment]
If you have are running the debugger, right after you click the login button it triggers this exception. The only way to counter this is to disable breaks for this type of exception. The first thing I did was to use a try catch block in the _Click event to try to grab the exception but it passes the statement, maybe due to the fact that the exception is C++ and I'm programming in C#. I did not try to simulate this with plain C++/CX.

I'm using the latest release of VS2017 btw.

**Comment by adiviness** [comment]
It is expected that the LoginAsync method will through an exception under certain circumstances. Visual Studio will tell you about these exceptions even if they will be properly taken care of by a try/catch block in the sdk. If you can tell Visual Studio to continue after the exception has occurred and you are able to log in then don't worry about it. If you can't log in after continuing past the exception than that could be a bug but right now this sounds like expected behavior.

**Comment by pdefreitas** [comment]
Yes login kinda works. It asks to input credentials again but the access token is defined and it is a valid one. Regarding the error I cant trace it well since I am using the nuget pkg.

**Comment by adiviness** [comment]
That's expected behavior for this scenario. The exception is thrown because the file we store the access token in does not exist. It then asks you for credentials to get a new access token.
